app/models/live_result.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class LiveResult < ApplicationRecord
+  has_many :live_attempts, -> { where(replaced_by: nil).order(:attempt_number) }
+
+  after_save :notify_users
+
+  belongs_to :registration
+
+  belongs_to :entered_by, class_name: 'User', foreign_key: 'entered_by_id'
```

Comment:
I think Rails is smart enough to infer the `foreign_key`, as long as it's "name of relation + _id"